### PR TITLE
Matlab add v2

### PR DESCRIPTION
**Pull request by matiarj**
Matlab scripts for pulling zmq data from pupil.

I had one version when I first started this that worked soleley inside matlab however it ends up slowing down matlab and will actually start to fall behind the pupil stream as some buffer I couldn't find was buffering old values.

### CODE REVIEW

File: pupil_remote/Matlab_Python/PupilNetwork2.m [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+eyeProperties.timerPeriod = 0.05; % in seconds 0.05 = 50 ms
+
+fopen(hUDP);
+eyeProperties.isOpen = true;
+
+tic; fread(hUDP); elapsed=toc;
+if elapsed>=eyeProperties.timerPeriod
+    sprintf('Timeout for PupilNetwork UDP (%.2f sec): Check UDP connection ',elapsed);
+    fclose(hUDP);
+    eyeProperties.isOpen = false;
```
Comment (papr) [inline pupil_remote/Matlab_Python/PupilNetwork2.m]: `isOpen` needs to be replaced with `isConnected`

File: pupil_remote/Matlab_Python/PupilNetwork2.m [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+else
+    eyeProperties.isConnected = true;
+end
+
```
Comment (papr) [inline pupil_remote/Matlab_Python/PupilNetwork2.m]: Running the `fread` test once does not work if Pupil Capture, and both matlab scripts (`pythonPupil.py`, `PupilNetwork2.m`) run on the same computer. A loop that reads multiple times to test the connection solves this problem.

**Comment by matiarj** [comment]
According to matlab documentation

>  Single-Precision Accuracy
> Similarly, there are gaps between any two single-precision numbers. If x has type single, eps(x) returns the distance between x and the next larger single-precision number. For example,
> 
> x = single(5);
> eps(x)
> returns
> 
> ans =
> 
>   single
> 
>   4.7684e-07

From what I understand regarding timestamps from pupil the smallest difference between two numbers should be limited by the sampling rate of the pupil hardware (120 Hz) which theoretically should be about 0.00833 seconds (8 ms) which should be captured by single precision. 

Please tell me if I'm not understanding this correctly.

File: pupil_remote/Matlab_Python/pythonPupil.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                conf_avg.dtype = 'uint8'
+
+                # issue arose where if you pulled timestamp again it would change the shape of timestamp1
+                # on the following iteration this avoids the shape change
+                # if x == 0:
+                timestamp = np.array([timestamp])
+                timestamp1 = timestamp.astype('float32')
+                timestamp1.dtype = 'uint8'
+                timestamp2 = np.squeeze(timestamp1)
+
```
Comment (matiarj) [inline pupil_remote/Matlab_Python/pythonPupil.py]: This is type casting.

File: pupil_remote/Matlab_Python/pupilRead.m [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+% return early if not connected
+if ~eyeProperties.isConnected,return;end
+
+% get data
+pkt = nan;
+num_iter = 0;
+max_iter = 20;
+fclose(hUDP);
+fopen(hUDP);
```
Comment (matiarj) [inline pupil_remote/Matlab_Python/pupilRead.m]: If i remember correctly if you don't do this buffers will start to hold old pupil gaze information

**Comment by willpatera** [comment]
@matiarj any updates on this branch. I'd love to see this plugin merged and part of the pupil-helpers repo after a bit of cleanup/response to @papr's points.

**Comment by matiarj** [comment]
I would love to however i'm currently in the midst of preparing for my
committee meeting for my phd. After I'm done writing and preparing I'll get
to these points.

@papr

1. A quick analysis of one of the sessions I found that the smallest time
interval found was slightly less than 8 ms which is captured by a single.
The reason I hesitate to move towards doubles is because by doing so we'd
essentially be doubling the amount of information we'd be sending over udp
and would risk possibly dropping packets or lag.
2. makes sense for longer sessions. The reason I reset the timebase is for
syncing purposes and for determining lag issues etc.

As far as documentation I'd have to sit down and spend a day looking
through the code I've provided here. This branch update is way different
from what I currently use, thus I'd have to reorient myself.

I hope to have my committee meeting set and done within the next month,
hopefully I can have this sorted out for you in the future. I may also have
a way to actually have matlab pull these values itself if you're interested
we could do a teamviewer session and I could show you. If this method works
it would completely remove the python middle man.

Again I apologize for the delayed response I've been swamped both in school
and in my personal life.

Best,
MJ

**Comment by willpatera** [comment]
@matiarj - I have discussed this PR and other PRs to the pupil-helpers repo and we have decided that bindings (e.g. Pascal bindings or MATLAB bindings or other language bindings) should exist as separate repositories, but will be linked in the pupil-helpers readme. The reasoning behind this decision, is due to the fact that the Pupil Labs team will not be able to maintain Pascal code (or Matlab code) and therefore it should not be included under this repo.

PRs to pupil-helpers repo should be revisions or new concise example scripts written in Python.

I will therefore close this PR and make an issue for myself to update the Readme with your repo.

**Pull request closed without merging**